worker/rhpv3.go

```diff
@@ -245,14 +254,15 @@ func dialTransport(ctx context.Context, siamuxAddr string, hostKey types.PublicK
 	}
 }
 
-func (p *transportPoolV3) withTransportV3(ctx context.Context, hostKey types.PublicKey, siamuxAddr string, fn func(context.Context, *transportV3) error) (err error) {
+func (p *transportPoolV3) withTransportV3(ctx context.Context, hostKey types.PublicKey, siamuxAddr, fallbackAddr string, fn func(context.Context, *transportV3) error) (err error) {
 	// Create or fetch transport.
 	p.mu.Lock()
 	t, found := p.pool[siamuxAddr]
 	if !found {
 		t = &transportV3{
-			hostKey:    hostKey,
-			siamuxAddr: siamuxAddr,
+			hostKey:      hostKey,
+			siamuxAddr:   siamuxAddr,
```

Comment:
It's probably a little tricky but perhaps `transportV3` could have

```go
mu                       sync.Mutex`
hostKey                  types.PublicKey
hostAddrs []string
```
where starting from the 2nd entry you just have the fallback addresses (which could be multiple)

Reply:
Hm yeah it's a little tricky. The underlying issue that's causing friction here is the fact that the contract metadata has the siamux address. I would remove it there, temporarily because we have to keep returning it from the API, and see what changes that induces. Among other changes, the host manager would become:

```go
HostManager interface {
    Host(ctx context.Context, hk types.PublicKey, fcid types.FileContractID) (Host,error)
}
```